infra/nightly-resources/index.js

```diff
@@ -1,46 +1,202 @@
-// Top-level load function for the main index page.
-function load() {
-  // data should be in the format of Array<{runMethod: String, benchmark: String, total_dyn_inst: Int, hyperfine: Array<{results: **hyperfine `--json` results**}>}>
-  console.log(data);
-
-  let container = document.getElementById("profile");
-  let parsed = [];
-
-  // aggregate benchmark runs into a list by their "benchmark" key
-  let groupedBy = {};
-  data.forEach((obj) => {
-    if(!groupedBy[obj.benchmark]) {
-      groupedBy[obj.benchmark] = [];
+async function getPreviousRuns() {
```

Comment:
nice- I would add a todo to fix this when we have a good way to get it cleanly

Reply:
That sounds like a good idea to me. I think this function needs a lot more documentation as well, so I'll add some in